internal/whisper/providers/awssm/client.go

```diff
@@ -6,24 +6,19 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
 	"github.com/google/uuid"
 )
 
 type client struct {
-	awsClient *secretsmanager.SecretsManager
+	awsClient *secretsmanager.Client
 }
 
 // New returns a client that fetches secrets from AWS Secrets Manager.
 func New() (*client, error) {
-	session, err := session.NewSession()
-	if err != nil {
-		return nil, err
-	}
-
-	c := secretsmanager.New(session)
+	opts := secretsmanager.Options{}
+	c := secretsmanager.New(opts)
```

Comment:
Looks like passing empty options prevents the client from configuring itself based on the environment whisper is running in. More details in other comment.

Reply:
LoadDefaultConfig docs: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/config#LoadDefaultConfig

NewFromConfig docs: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/secretsmanager#NewFromConfig

Reply:
Yep, you're right. Fixing now.

Are you okay with a `context.TODO()` in the provider constructor or should I go ahead and plumb through a proper context from the resolver?